doc/sphinx/proof-engine/vernacular-commands.rst

```diff
@@ -220,63 +220,79 @@ Requests to the environment
    .. seealso:: Section :ref:`performingcomputations`.
 
 
-.. cmd:: Print Assumptions @qualid
+.. cmd:: Print Assumptions @smart_qualid
 
-   This commands display all the assumptions (axioms, parameters and
-   variables) a theorem or definition depends on. Especially, it informs
-   on the assumptions with respect to which the validity of a theorem
-   relies.
+   Displays all the assumptions (axioms, parameters and
+   variables) a theorem or definition depends on.
 
-   .. cmdv:: Print Opaque Dependencies @qualid
-      :name: Print Opaque Dependencies
+   .. exn:: Closed under the global context
 
-      Displays the set of opaque constants :n:`@qualid` relies on in addition to
-      the assumptions.
+      The theorem or definition doesn't depend on any assumptions.
 
-   .. cmdv:: Print Transparent Dependencies @qualid
-      :name: Print Transparent Dependencies
+.. cmd:: Print Opaque Dependencies @smart_qualid
 
-      Displays the set of transparent constants :n:`@qualid` relies on
-      in addition to the assumptions.
+   Displays the assumptions and opaque constants that :n:`@smart_qualid` depends on.
 
-   .. cmdv:: Print All Dependencies @qualid
-      :name: Print All Dependencies
+.. cmd:: Print Transparent Dependencies @smart_qualid
 
-      Displays all assumptions and constants :n:`@qualid` relies on.
+   Displays the assumptions and  transparent constants that :n:`@smart_qualid` depends on.
 
+.. cmd:: Print All Dependencies @smart_qualid
 
-.. cmd:: Search @qualid
+   Displays all the assumptions and constants :n:`@smart_qualid` relies on.
 
-   This command displays the name and type of all objects (hypothesis of
-   the current goal, theorems, axioms, etc) of the current context whose
-   statement contains :n:`@qualid`. This command is useful to remind the user
-   of the name of library lemmas.
 
-   .. exn:: The reference @qualid was not found in the current environment.
+.. cmd:: Search {+ @search_item } {? {| inside | outside } {+ @qualid } }
 
-      There is no constant in the environment named qualid.
+   .. insertprodn search_item search_item
+
+   .. prodn::
+      search_item ::= {? - } @string {? % @ident }
+      | {? - } @one_term
+
+   This command displays the name and type of objects (hypothesis of
+   the current goal, theorems, axioms, etc) in the current context.
+   This command is useful to remind the user of the names of library lemmas.
 
-   .. cmdv:: Search @string
+   .. todo above: "current context" meaning on the current goal or the first goal?
+      below: adapted the old wording, doesn't seem consistent
+      also will "invocation-of-tactics" cover the command case, or will there be a new section?
```

Comment:
I think I've answered this question in my suggestion above + a previous comment of mine regarding the fact that *goal selectors* have the same meaning for tactics and commands.